### DISCUSSION TITLE
Interactive tutorials (with learnr) within Quarto website

### DISCUSSION THREAD

**Discussion by graveja0** [quarto-dev/quarto-cli#926] [Q&A]
I'm hoping to design a Quarto website for some courses and workshops I am developing this year. However, in browsing through it seems I can create a website with some material (e.g., presentations, static documents, and Shiny apps) but not others (e.g., interactive tutorials created with learnr package). Am I missing something within the current functionality, or is the integration of interactive tutorials in the works? Thanks!

**Comment by jjallaire** [comment]
We aren't planning on including interactive tutorials on the short or intermediate term, but I'm sure we will in the fullness of time.

That said, R Markdown / distill / blogdown / pkgdown sites can't really include learnr tutorials (or even shiny apps) directly. Rather, they would use an iframe to embed it. You can do the same thing with Quarto.

**Reply by JeffreyRStevens** [reply]
@jjallaire , is there a way to at least render a learnr tutorial that is in a website's directory structure? I'm fine publishing it on shinyapps.io, but currently I can't render a tutorial that resides in a Quarto website directory. When I try, I receive the following error:
```
Error in shiny_prerendered_html(input_rmd, render_args) : 
  Prerendered HTML file not found at exercises/01_introduction/01_introduction.html
Calls: .main ... <Anonymous> -> shiny_prerendered_app -> shiny_prerendered_html
Execution halted
```
When I render outside of the Quarto website project, it renders fine.

**Reply by JeffreyRStevens** [reply]
Looks like if I use `rmarkdown::run()` it renders properly. But is there a way to set up the YAML headers or a `_quarto.yml` file so that clicking **Run Document** or pressing ctrl+shift+k will make a learnr tutorial render properly?
